File: src/main/java/com/specforge/catalog/SpecStatus.java
package com.specforge.catalog;

/**
 * The lifecycle of a specification document: Draft, In Review, Changes Requested, Approved,
 * Implemented. The legal transitions between these are enforced in one place inside this module,
 * not by whoever happens to be setting the field.
 */
public enum SpecStatus {
    DRAFT,
    IN_REVIEW,
    CHANGES_REQUESTED,
    APPROVED,
    IMPLEMENTED
}

File: frontend/src/lib/specStatus.ts
import type { SpecStatus } from '../api/catalog';

export const SPEC_STATUSES: SpecStatus[] = ['DRAFT', 'IN_REVIEW', 'CHANGES_REQUESTED', 'APPROVED', 'IMPLEMENTED'];

const LABEL: Record<SpecStatus, string> = {
  DRAFT: 'Draft',
  IN_REVIEW: 'In review',
  CHANGES_REQUESTED: 'Changes requested',
  APPROVED: 'Approved',
  IMPLEMENTED: 'Implemented',
};

export function specStatusLabel(status: SpecStatus): string {
  return LABEL[status];
}
